adaptdl/adaptdl/speedup.py

```diff
@@ -283,18 +316,21 @@ def _rmse(pred, true):
     return np.sqrt(((pred - true) ** 2).mean())
 
 
-def _obj_fn(params, nodes, replicas, local_bsz, step_time, step_time_compute):
+def _obj_fn(params, nodes, replicas, local_bsz, grad_acc_steps,
+            step_time, step_time_compute, grad_acc_time):
     params = Params(*params)
-    log_pred_step_time, pred_step_time_compute, _ = \
-        _predict_log(params, nodes, replicas, local_bsz)
+    log_pred_step_time, pred_step_time_compute, _, pred_grad_acc_time = \
+        _predict_log(params, nodes, replicas, local_bsz, grad_acc_steps)
     # Error of total step time predictions.
     err1 = _rmse(log_pred_step_time, np.log(step_time))
     # Error of compute time predictions.
     err2 = _rmse(np.log(pred_step_time_compute), np.log(step_time_compute))
+    # Error of gradient accumulation
+    err3 = _rmse(np.log(pred_grad_acc_time), np.log(grad_acc_time))
```

Comment:
We have prior knowledge that the `compute_time` should be the same as (at least similar to) the `step_time` for accumulated steps. So the objective function can be simplified to just two error terms, one for the non-accumulated step time, and one for the accumulated (combined with `compute_time`) step time. That should make this function easier to fit.

Reply:
Removed